### ISSUE TITLE
[BUG] Running cleanup and or cleanup_all fails with 'Failed to raise an exception: END_OF_STACK'

### ISSUE THREAD

**Issue by agentpoyo** [cncf/cnf-conformance#351]
**Describe the bug**
When running the latest stable binary release v0.0.9 following the curl install instructions and after running sample coredns test, running the cleanup and or cleanup_all command fails with `Failed to raise an exception: END_OF_STACK`

**Detailed Description**
`cnf-conformance cleanup_all` which should cleanup all test deploys fails. 

**To Reproduce**

1. Follow the curl install instructions to pull latest stable v0.0.9 release.
2. Run cnf-conformance steps to run tests
3. Now attempt to cleanup, see below on outputs:

Version Info Verify
```
$ cnf-conformance version
CNF Conformance version: v0.0.9
```
Help Output for Cleanup Commands Available
```
$ cnf-conformance help | grep clean
cleanup                                                 Cleans up the CNF Conformance test suite, the K8s cluster, and upstream projects
samples_cleanup                                         Cleans up the CNF Conformance sample projects
tools_cleanup                                           
cleanup_all                                             
results_yml_cleanup                                     
cnf_cleanup                                             
sample_coredns_cleanup                                  
cleanup_sample_coredns                                  
bad_helm_cnf_cleanup                                    
sample_privileged_cnf_whitelisted_cleanup               
sample_privileged_cnf_non_whitelisted_cleanup           
sample_coredns_bad_liveness_cleanup                     
sample_coredns_source_cleanup                           
sample_generic_cnf_cleanup                              
helm_local_cleanup                                      Cleans up helm 3.1.1
sonobuoy_cleanup                                        Cleans up Sonobuoy
```

Verify test deploys:
```
helm list --all
NAME      	NAMESPACE	REVISION	UPDATED                                	STATUS  	CHART         	APP VERSION
chaos-mesh	default  	1       	2020-08-10 16:43:32.953552562 +0000 UTC	deployed	chaos-mesh-dev	1.0        
coredns   	default  	1       	2020-08-10 16:39:32.627482605 +0000 UTC	deployed	coredns-1.13.2	1.7.0
```
```
kubectl get pods
NAME                                        READY   STATUS    RESTARTS   AGE
chaos-controller-manager-7f9bbd476f-992sw   1/1     Running   0          9m50s
chaos-daemon-j94bs                          1/1     Running   0          9m50s
coredns-coredns-5f8c9f75b7-5jrv7            1/1     Running   0          10m
```

Running the cleanup_all
```
$ cnf-conformance cleanup_all
Failed to raise an exception: END_OF_STACK
[0x7fd13368ee76] ???
[0x7fd13346389b] ???
[0x7fd133464185] ???
[0x7fd133470f34] ???
[0x7fd13346925c] ???
[0x7fd133468a79] ???
[0x7fd1334b5df5] ???
[0x7fd13348415c] ???
[0x7fd1334668f7] ???
[0x7fd13390a054] ???
```

Specifying the config fails:
```
$ cnf-conformance cleanup cnf-config=./cnfs/coredns/cnf-conformance.yml
Failed to raise an exception: END_OF_STACK
[0x7fadf74c0e76] ???
[0x7fadf729589b] ???
[0x7fadf7296185] ???
[0x7fadf72a2f34] ???
[0x7fadf729b25c] ???
[0x7fadf729aa79] ???
[0x7fadf72e7df5] ???
[0x7fadf72b615c] ???
[0x7fadf72988f7] ???
[0x7fadf773c054] ???
```

**Expected behavior**
We should expect that all deploys including the chaosmesh is removed/uninstalled.

**Workaround**
Manually clean the cluster with `helm delete name`:
```
$ helm list
NAME      	NAMESPACE	REVISION	UPDATED                                	STATUS  	CHART         	APP VERSION
chaos-mesh	default  	1       	2020-08-10 16:43:32.953552562 +0000 UTC	deployed	chaos-mesh-dev	1.0        
coredns   	default  	1       	2020-08-10 16:39:32.627482605 +0000 UTC	deployed	coredns-1.13.2	1.7.0  
    
$ helm delete coredns
release "coredns" uninstalled

$ helm delete chaos-mesh
release "chaos-mesh" uninstalled

$ kubectl get pods
No resources found.
```

Note: Using the lastest master branch, we see the following when attempting to run `cleanup_all` after building cnf-conformance from source:
```
 ./cnf-conformance cleanup_all
lib/sam/src/sam/task.cr:99:7 in 'find!'
lib/sam/src/sam/task.cr:57:7 in 'invoke'
lib/sam/src/sam/task.cr:39:29 in 'call'
lib/sam/src/sam.cr:43:5 in 'invoke'
lib/sam/src/sam.cr:77:7 in 'process_tasks'
lib/sam/src/sam.cr:56:7 in 'help'
src/cnf-conformance.cr:54:1 in '__crystal_main'
/usr/share/crystal/src/crystal/main.cr:106:5 in 'main_user_code'
/usr/share/crystal/src/crystal/main.cr:92:7 in 'main'
/usr/share/crystal/src/crystal/main.cr:115:3 in 'main'
__libc_start_main
_start
???
Task cleanup_samples was not found
```

But running the simple `cleanup` appears to cleanup everything:
```
./cnf-conformance cleanup
I, [2020-08-10 17:34:14 +00:00 #4277]  INFO -- cnf-conformance-verbose: cleanup config: #<Totem::Config @config_paths=["sample-cnfs/sample-coredns-cnf"], @config_name="cnf-conformance", @config_type="yml", @config_envs="["development", "test", "production"]", @config_env="", @key_delimiter=".", @automatic_env=false, @env_prefix=nil, @aliases={}, @overrides={}, @config={"helm_directory" => "helm_chart", "git_clone_url" => nil, "install_script" => nil, "release_name" => "coredns", "deployment_name" => "coredns-coredns", "deployment_label" => "k8s-app", "service_name" => "coredns-coredns", "application_deployment_names" => ["coredns-coredns"], "docker_repository" => "coredns/coredns", "helm_repository" => {"name" => "stable", "repo_url" => "https://kubernetes-charts.storage.googleapis.com"}, "helm_chart" => "stable/coredns", "helm_chart_container_name" => "coredns", "rolling_update_tag" => "1.6.7", "white_list_helm_chart_container_names" => ["falco", "node-cache", "nginx", "coredns", "calico-node", "kube-proxy", "nginx-proxy"]}, @env={}, @kvstores={}, @defaults={}>
I, [2020-08-10 17:34:14 +00:00 #4277]  INFO -- cnf-conformance-verbose: 
I, [2020-08-10 17:34:18 +00:00 #4277]  INFO -- cnf-conformance-verbose: release "coredns" uninstalled

Successfully cleaned up coredns
```
The remaining chaos-mesh output after the output that coredns was removed successfully:
```
I, [2020-08-10 17:34:18 +00:00 #4277]  INFO -- cnf-conformance-verbose: cleanup config: #<Totem::Config @config_paths=["sample-cnfs/sample_coredns"], @config_name="cnf-conformance", @config_type="yml", @config_envs="["development", "test", "production"]", @config_env="", @key_delimiter=".", @automatic_env=false, @env_prefix=nil, @aliases={}, @overrides={}, @config={"helm_directory" => "chart", "git_clone_url" => nil, "install_script" => "chart", "release_name" => "coredns", "deployment_name" => "coredns-coredns", "deployment_label" => "k8s-app", "service_name" => "coredns-coredns", "application_deployment_names" => ["coredns"], "helm_chart" => "stable/coredns", "helm_chart_container_name" => "coredns", "rolling_update_tag" => "1.6.7", "white_list_helm_chart_container_names" => []}, @env={}, @kvstores={}, @defaults={}>
I, [2020-08-10 17:34:18 +00:00 #4277]  INFO -- cnf-conformance-verbose: cleanup config: #<Totem::Config @config_paths=["sample-cnfs/sample-bad_helm_coredns-cnf"], @config_name="cnf-conformance", @config_type="yml", @config_envs="["development", "test", "production"]", @config_env="", @key_delimiter=".", @automatic_env=false, @env_prefix=nil, @aliases={}, @overrides={}, @config={"helm_directory" => "chart", "git_clone_url" => nil, "install_script" => nil, "release_name" => "bad-helm-coredns-coredns", "deployment_name" => "bad-helm-coredns-coredns", "deployment_label" => "k8s-app", "service_name" => "bad-helm-coredns-coredns", "application_deployment_names" => ["bad-helm-coredns"], "helm_chart" => "stable/coredns", "helm_chart_container_name" => "coredns", "white_list_helm_chart_container_names" => ["falco", "nginx", "coredns", "calico-node", "kube-proxy", "nginx-proxy"]}, @env={}, @kvstores={}, @defaults={}>
I, [2020-08-10 17:34:18 +00:00 #4277]  INFO -- cnf-conformance-verbose: cleanup config: #<Totem::Config @config_paths=["sample-cnfs/sample_privileged_cnf"], @config_name="cnf-conformance", @config_type="yml", @config_envs="["development", "test", "production"]", @config_env="", @key_delimiter=".", @automatic_env=false, @env_prefix=nil, @aliases={}, @overrides={}, @config={"helm_directory" => "chart", "git_clone_url" => nil, "install_script" => nil, "release_name" => "privileged-coredns", "deployment_name" => "privileged-coredns-coredns", "deployment_label" => "k8s-app", "service_name" => "privileged-coredns-coredns", "application_deployment_names" => ["privileged-coredns-coredns"], "helm_chart" => "stable/coredns", "helm_chart_container_name" => "privileged-coredns-coredns", "white_list_helm_chart_container_names" => []}, @env={}, @kvstores={}, @defaults={}>
I, [2020-08-10 17:34:18 +00:00 #4277]  INFO -- cnf-conformance-verbose: cleanup config: #<Totem::Config @config_paths=["sample-cnfs/sample_whitelisted_privileged_cnf"], @config_name="cnf-conformance", @config_type="yml", @config_envs="["development", "test", "production"]", @config_env="", @key_delimiter=".", @automatic_env=false, @env_prefix=nil, @aliases={}, @overrides={}, @config={"helm_directory" => "chart", "git_clone_url" => nil, "install_script" => nil, "release_name" => "privileged-coredns", "deployment_name" => "privileged-coredns-coredns", "deployment_label" => "k8s-app", "service_name" => "privileged-coredns-coredns", "application_deployment_names" => ["privileged-coredns-coredns"], "helm_chart" => "stable/coredns", "helm_chart_container_name" => "privileged-coredns-coredns", "white_list_helm_chart_container_names" => ["coredns"]}, @env={}, @kvstores={}, @defaults={}>
I, [2020-08-10 17:34:18 +00:00 #4277]  INFO -- cnf-conformance-verbose: cleanup config: #<Totem::Config @config_paths=["sample-cnfs/sample_coredns_bad_liveness"], @config_name="cnf-conformance", @config_type="yml", @config_envs="["development", "test", "production"]", @config_env="", @key_delimiter=".", @automatic_env=false, @env_prefix=nil, @aliases={}, @overrides={}, @config={"helm_directory" => "chart", "git_clone_url" => nil, "install_script" => "chart", "release_name" => "bad-liveness", "deployment_name" => "bad-liveness-coredns", "deployment_label" => "k8s-app", "service_name" => "bad-liveness-coredns", "application_deployment_names" => ["bad-liveness-coredns"], "helm_chart" => "stable/coredns", "helm_chart_container_name" => "bad-liveness-coredns", "white_list_helm_chart_container_names" => []}, @env={}, @kvstores={}, @defaults={}>
I, [2020-08-10 17:34:18 +00:00 #4277]  INFO -- cnf-conformance-verbose: cleanup config: #<Totem::Config @config_paths=["sample-cnfs/sample-coredns-cnf-source"], @config_name="cnf-conformance", @config_type="yml", @config_envs="["development", "test", "production"]", @config_env="", @key_delimiter=".", @automatic_env=false, @env_prefix=nil, @aliases={}, @overrides={}, @config={"helm_directory" => "helm_chart/coredns", "git_clone_url" => "https://github.com/coredns/coredns.git", "install_script" => "coredns/Makefile", "release_name" => "coredns", "deployment_name" => "coredns-coredns", "deployment_label" => "k8s-app", "service_name" => "coredns-coredns", "application_deployment_names" => ["coredns-coredns"], "helm_chart" => "stable/coredns", "helm_chart_container_name" => "coredns", "white_list_helm_chart_container_names" => ["falco", "nginx", "coredns", "calico-node", "kube-proxy", "nginx-proxy"]}, @env={}, @kvstores={}, @defaults={}>
I, [2020-08-10 17:34:18 +00:00 #4277]  INFO -- cnf-conformance-verbose: cleanup config: #<Totem::Config @config_paths=["sample-cnfs/sample-generic-cnf"], @config_name="cnf-conformance", @config_type="yml", @config_envs="["development", "test", "production"]", @config_env="", @key_delimiter=".", @automatic_env=false, @env_prefix=nil, @aliases={}, @overrides={}, @config={"helm_directory" => "helm_chart", "git_clone_url" => "https://github.com/coredns/coredns.git", "install_script" => "cnfs/coredns/Makefile", "release_name" => "coredns", "deployment_name" => "coredns-coredns", "deployment_label" => "k8s-app", "service_name" => "coredns-coredns", "application_deployment_names" => ["coredns-coredns"], "docker_repository" => "coredns/coredns", "helm_repository" => {"name" => "stable", "repo_url" => "https://kubernetes-charts.storage.googleapis.com"}, "helm_chart" => "stable/coredns", "helm_chart_container_name" => "coredns", "white_list_helm_chart_container_names" => ["falco", "nginx", "coredns", "calico-node", "kube-proxy", "nginx-proxy"]}, @env={}, @kvstores={}, @defaults={}>
Error from server (NotFound): error when deleting "https://raw.githubusercontent.com/pingcap/chaos-mesh/master/manifests/crd.yaml": customresourcedefinitions.apiextensions.k8s.io "iochaos.chaos-mesh.org" not found
Error from server (NotFound): error when deleting "https://raw.githubusercontent.com/pingcap/chaos-mesh/master/manifests/crd.yaml": customresourcedefinitions.apiextensions.k8s.io "kernelchaos.chaos-mesh.org" not found
Error from server (NotFound): error when deleting "https://raw.githubusercontent.com/pingcap/chaos-mesh/master/manifests/crd.yaml": customresourcedefinitions.apiextensions.k8s.io "networkchaos.chaos-mesh.org" not found
Error from server (NotFound): error when deleting "https://raw.githubusercontent.com/pingcap/chaos-mesh/master/manifests/crd.yaml": customresourcedefinitions.apiextensions.k8s.io "podchaos.chaos-mesh.org" not found
Error from server (NotFound): error when deleting "https://raw.githubusercontent.com/pingcap/chaos-mesh/master/manifests/crd.yaml": customresourcedefinitions.apiextensions.k8s.io "stresschaos.chaos-mesh.org" not found
Error from server (NotFound): error when deleting "https://raw.githubusercontent.com/pingcap/chaos-mesh/master/manifests/crd.yaml": customresourcedefinitions.apiextensions.k8s.io "timechaos.chaos-mesh.org" not found
```
Verifying deploys are cleaned up:
```
$ helm list 
NAME	NAMESPACE	REVISION	UPDATED	STATUS	CHART	APP VERSION

$ kubectl get pods
No resources found in default namespace.
```

**Workstation (please complete the following information):**
 - OS = Linux
 - Distro = Ubuntu
 - Version = 18.04
 - Architecture = amd64/x86

**Comment by lixuna** [comment]
This issue seems related to #369

**Comment by wavell** [comment]
![image](https://user-images.githubusercontent.com/29548/97356479-52cf9a00-1866-11eb-8f74-3bf656c50586.png)
Might have been fixed from #437

**Comment by nupejosh** [comment]
Followed instructions and got desired behavior. See screenshots of steps followed below

![Screen Shot 2020-10-28 at 1 29 10 PM](https://user-images.githubusercontent.com/3164983/97481099-5163a780-1922-11eb-9617-c3ebb9353cb2.png)
![Screen Shot 2020-10-28 at 1 29 18 PM](https://user-images.githubusercontent.com/3164983/97481100-5163a780-1922-11eb-977c-80ab705024df.png)
![Screen Shot 2020-10-28 at 1 29 33 PM](https://user-images.githubusercontent.com/3164983/97481104-5163a780-1922-11eb-91aa-61d9a0cbdcce.png)
![Screen Shot 2020-10-28 at 1 29 51 PM](https://user-images.githubusercontent.com/3164983/97481095-50cb1100-1922-11eb-9bca-ddd4bff0a88a.png)

**Comment by lixuna** [comment]
@wavell @nupejosh @agentpoyo what is your estimate in points for this bugfix, noting that it might have been fixed from #437?

1 or 2 points?

**Comment by agentpoyo** [comment]
2pts
